crates/p2p/examples/peer.rs

```diff
@@ -0,0 +1,180 @@
+#![deny(rust_2018_idioms)]
+
+use std::sync::Arc;
+use std::{path::Path, time::Duration};
+
+use clap::Parser;
+use libp2p::Multiaddr;
+use libp2p::{identity::Keypair, PeerId};
+use p2p::Peers;
+use p2p_proto as proto;
+use proto::sync::{BlockBodies, StateDiffs};
+use serde_derive::Deserialize;
+use stark_hash::Felt;
+use tokio::sync::RwLock;
+use zeroize::Zeroizing;
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[clap(long, value_parser, env = "IDENTITY_CONFIG_FILE")]
+    identity_config_file: Option<std::path::PathBuf>,
+    #[clap(long, value_parser, env = "LISTEN_ON")]
+    listen_on: Multiaddr,
+    #[clap(long, value_parser, env = "BOOTSTRAP_ADDRESSES")]
+    bootstrap_addresses: Vec<Multiaddr>,
+    #[clap(long)]
+    emit_events: bool,
+}
+
+#[derive(Clone, Deserialize)]
+struct IdentityConfig {
+    pub private_key: String,
+}
+
+impl IdentityConfig {
+    pub fn from_file(path: &Path) -> anyhow::Result<Self> {
+        Ok(serde_json::from_str(&std::fs::read_to_string(path)?)?)
+    }
+}
+
+impl zeroize::Zeroize for IdentityConfig {
+    fn zeroize(&mut self) {
+        self.private_key.zeroize()
+    }
+}
+
+const PERIODIC_STATUS_INTERVAL: Duration = Duration::from_secs(30);
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    if std::env::var_os("RUST_LOG").is_none() {
+        std::env::set_var("RUST_LOG", "info");
+    }
+
+    setup_tracing();
+
+    let args = Args::parse();
+    let keypair = match &args.identity_config_file {
```

Comment:
I was wondering if the secret in the keypair is zeroized on drop and it is :sunglasses: